paint/PAINT_ShapeTool.cpp

```diff
@@ -1,16 +1,51 @@
 #include "PAINT_pch.h"
 #include "PAINT_ShapeTool.h"
+#include "PAINT_Brush.h"
+#include "GFX_Rectangle.h"
 
 using namespace paint;
 
-ShapeTool::ShapeTool()
+ShapeTool::ShapeTool(SDL_Renderer* renderer, SDL_Texture* texture)
 	: activeShape_(nullptr)
+	, renderer_(renderer)
+	, texture_(texture)
 {
-	
+	activeBrush_ = std::make_shared<Brush>(0);
 }
 
-void ShapeTool::setActiveShape(Shape* shape)
+void ShapeTool::setActiveShape(std::shared_ptr<Shape> shape)
 {
-	activeShape_ = shape;
+	activeShape_ = std::move(shape);
 }
 
+void ShapeTool::toolFunction(win::Coords mouseCoords, win::Coords prevMouseCoords, win::Coords startCoords, gfx::Rectangle refRect)
+{
+	clearLines();
+	const win::Coords rel = { mouseCoords.x - refRect.x, mouseCoords.y - refRect.y };
+	if (rel.x < 0)
```

Comment:
Bracing style for `if` is wrong. Should be the (admittedly unpleasant but spatially superior) K&R style.
**MUST** as it's the Coding Standard.

Reply:
done